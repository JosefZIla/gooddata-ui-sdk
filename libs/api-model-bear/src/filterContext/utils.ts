// (C) 2019-2023 GoodData Corporation
import omitBy from "lodash/omitBy";
import isUndefined from "lodash/isUndefined";
import { GdcFilterContext } from "./GdcFilterContext";
import isDateFilter = GdcFilterContext.isDateFilter;
import isAttributeFilter = GdcFilterContext.isAttributeFilter;

function sanitizeDateFilter(filter: GdcFilterContext.IDateFilter): GdcFilterContext.IDateFilter {
    const {
        dateFilter: { from, to, type, granularity },
        dateFilter,
    } = filter;
    const sanitizedProperties = omitBy(dateFilter, isUndefined);

    return {
        dateFilter: {
            type,
            granularity,
            ...sanitizedProperties,
            ...(from === undefined ? {} : { from: String(from) }),
            ...(to === undefined ? {} : { to: String(to) }),
        },
    };
}

function sanitizeAttributeFilter(
    filter: GdcFilterContext.IAttributeFilter,
): GdcFilterContext.IAttributeFilter {
    const {
        attributeFilter: { displayForm, negativeSelection, attributeElements, selectionMode },
    } = filter;

    return {
        attributeFilter: {
            displayForm,
            negativeSelection,
            attributeElements,
            selectionMode,
        },
    };
}

/**
 * @public
 */
export function sanitizeFiltersForExport(
    filters: GdcFilterContext.FilterContextItem[],
): GdcFilterContext.FilterContextItem[] {
    return filters.map((filter: GdcFilterContext.FilterContextItem): GdcFilterContext.FilterContextItem => {
        if (isDateFilter(filter)) {
            return sanitizeDateFilter(filter);
        }
        if (isAttributeFilter(filter)) {
            return sanitizeAttributeFilter(filter);
        }
        return filter;
    });
}
