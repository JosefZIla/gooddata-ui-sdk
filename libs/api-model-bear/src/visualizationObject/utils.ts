// (C) 2007-2022 GoodData Corporation
import {
    IVisualizationObjectContent,
    IBucket,
    BucketItem,
    IVisualizationAttributeContent,
    IMeasureContent,
    IVisualizationObjectMeasureDefinition,
    isVisualizationObjectMeasureDefinition,
    VisualizationObjectFilter,
    VisualizationObjectAttributeFilter,
    isVisualizationObjectAttribute,
    isVisualizationObjectMeasure,
    isVisualizationObjectAttributeFilter,
    isVisualizationObjectPositiveAttributeFilter,
} from "./GdcVisualizationObject.js";
import { IObjUriQualifier } from "../base/GdcTypes.js";

function getAttributesInBucket(bucket: IBucket): IVisualizationAttributeContent[] {
    return bucket.items.reduce((list: IVisualizationAttributeContent[], bucketItem: BucketItem) => {
        if (isVisualizationObjectAttribute(bucketItem)) {
            list.push(bucketItem.visualizationAttribute);
        }
        return list;
    }, []);
}

function getAttributes(mdObject: IVisualizationObjectContent): IVisualizationAttributeContent[] {
    const buckets = mdObject.buckets;
    return buckets.reduce((categoriesList: IVisualizationAttributeContent[], bucket: IBucket) => {
        categoriesList.push(...getAttributesInBucket(bucket));
        return categoriesList;
    }, []);
}

function getMeasuresInBucket(bucket: IBucket): IMeasureContent[] {
    return bucket.items.reduce((list: IMeasureContent[], bucketItem: BucketItem) => {
        if (isVisualizationObjectMeasure(bucketItem)) {
            list.push(bucketItem.measure);
        }
        return list;
    }, []);
}

function getDefinition(
    measure: IMeasureContent,
): IVisualizationObjectMeasureDefinition["measureDefinition"] | undefined {
    return isVisualizationObjectMeasureDefinition(measure.definition)
        ? measure.definition.measureDefinition
        : undefined;
}

function getMeasures(mdObject: IVisualizationObjectContent): IMeasureContent[] {
    const buckets = mdObject.buckets;
    return buckets.reduce((measuresList: IMeasureContent[], bucket: IBucket) => {
        measuresList.push(...getMeasuresInBucket(bucket));
        return measuresList;
    }, []);
}

function getMeasureFilters(measure: IMeasureContent): VisualizationObjectFilter[] {
    return getDefinition(measure)?.filters ?? [];
}

function getMeasureAttributeFilters(measure: IMeasureContent): VisualizationObjectAttributeFilter[] {
    return getMeasureFilters(measure).filter(isVisualizationObjectAttributeFilter);
}

function getAttributeFilters(mdObject: IVisualizationObjectContent): VisualizationObjectAttributeFilter[] {
    return getMeasures(mdObject).reduce(
        (filters: VisualizationObjectAttributeFilter[], measure: IMeasureContent) => {
            filters.push(...getMeasureAttributeFilters(measure));
            return filters;
        },
        [],
    );
}

function getAttributeFilterDisplayForm(measureFilter: VisualizationObjectAttributeFilter): string {
    return isVisualizationObjectPositiveAttributeFilter(measureFilter)
        ? (measureFilter.positiveAttributeFilter.displayForm as IObjUriQualifier).uri
        : (measureFilter.negativeAttributeFilter.displayForm as IObjUriQualifier).uri;
}

/**
 * @public
 */
export function getAttributesDisplayForms(mdObject: IVisualizationObjectContent): string[] {
    const attributesDfs = getAttributes(mdObject).map(
        (attribute: IVisualizationAttributeContent) => (attribute.displayForm as IObjUriQualifier).uri,
    );
    const attrMeasureFilters = getAttributeFilters(mdObject);
    const attrMeasureFiltersDfs = attrMeasureFilters.map(getAttributeFilterDisplayForm);
    return [...attrMeasureFiltersDfs, ...attributesDfs];
}
