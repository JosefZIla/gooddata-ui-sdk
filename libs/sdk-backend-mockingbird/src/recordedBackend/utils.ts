// (C) 2019-2021 GoodData Corporation

/**
 * @internal
 */
export function identifierToRecording(id: string): string {
    return id.replace(/\./g, "_");
}
