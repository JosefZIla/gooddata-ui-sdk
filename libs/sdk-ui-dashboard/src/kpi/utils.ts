// (C) 2020-2021 GoodData Corporation
import isNil from "lodash/isNil";
import isNumber from "lodash/isNumber";
import { IDataSeries } from "@gooddata/sdk-ui";
/**
 * @internal
 */
export function getKpiSeriesResult(series: IDataSeries | undefined): number | null {
    if (!series) {
        return null;
    }

    const value = series.dataPoints()[0].rawValue;

    if (isNil(value)) {
        return null;
    }

    if (isNumber(value)) {
        return value;
    }

    return Number.parseFloat(value);
}
