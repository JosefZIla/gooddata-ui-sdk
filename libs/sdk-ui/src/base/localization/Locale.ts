// (C) 2007-2018 GoodData Corporation

/**
 * Language codes that can be used with GoodData.
 *
 * @public
 */
export type ILocale =
    | "en-US"
    | "de-DE"
    | "es-ES"
    | "fr-FR"
    | "ja-JP"
    | "nl-NL"
    | "pt-BR"
    | "pt-PT"
    | "zh-Hans";

/**
 * Default value for {@link ILocale}.
 *
 * @public
 */
export const DefaultLocale: ILocale = "en-US";
