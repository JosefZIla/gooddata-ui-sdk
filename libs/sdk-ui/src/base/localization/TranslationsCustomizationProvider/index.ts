// (C) 2021 GoodData Corporation
export {
    ITranslationsCustomizationContextProviderProps,
    TranslationsCustomizationContextProvider,
    withTranslationsCustomization,
} from "./Context";
export {
    ITranslationsCustomizationProviderProps,
    TranslationsCustomizationProvider,
} from "./TranslationsCustomizationProvider";
export { pickCorrectInsightWording } from "./utils";
