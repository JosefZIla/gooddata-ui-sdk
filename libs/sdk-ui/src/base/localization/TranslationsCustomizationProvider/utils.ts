// (C) 2007-2021 GoodData Corporation
import { IWorkspaceSettings } from "@gooddata/sdk-backend-spi";

const getNewKey = (key: string, stringToRemove: string) => key.replace(stringToRemove, "");

/**
 * @beta
 */
export const pickCorrectInsightWording = (
    translations: Record<string, string>,
    settings?: IWorkspaceSettings,
): Record<string, string> => {
    const isEnabledInsightToReport = !!settings?.enableInsightToReport;
    const modifiedTranslations = {};
    Object.keys(translations).forEach((key) => {
        if (key.includes("|report") || key.includes("|insight")) {
            const newKey = getNewKey(key, isEnabledInsightToReport ? "|report" : "|insight");
            modifiedTranslations[newKey] = translations[key];
        }
    });
    return {
        ...translations,
        ...modifiedTranslations,
    };
};
