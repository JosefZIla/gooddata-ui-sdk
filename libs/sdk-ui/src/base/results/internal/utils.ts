// (C) 2019-2022 GoodData Corporation

import { IDataView } from "@gooddata/sdk-backend-spi";
import {
    IMeasureDescriptor,
    IDimensionItemDescriptor,
    IMeasureGroupDescriptor,
    IResultHeader,
} from "@gooddata/sdk-model";

/**
 * @internal
 */
export function dataViewHeaders(dataView: IDataView, dimIdx: number): IResultHeader[][] {
    return dataView.headerItems[dimIdx] ?? [];
}

/**
 * @internal
 */
export function dataViewDimensionItems(dataView: IDataView, dimIdx: number): IDimensionItemDescriptor[] {
    const dimensionDescriptor = dataView.result.dimensions[dimIdx];

    return dimensionDescriptor ? dimensionDescriptor.headers : [];
}

/**
 * @internal
 */
export function measureGroupItems(measureGroup: IMeasureGroupDescriptor): IMeasureDescriptor[] {
    return measureGroup.measureGroupHeader.items;
}

/**
 * @internal
 */
export function measureFormat(measureDescriptor: IMeasureDescriptor): string {
    return measureDescriptor.measureHeaderItem.format;
}

/**
 * @internal
 */
export function measureName(measureDescriptor: IMeasureDescriptor): string {
    return measureDescriptor.measureHeaderItem.name;
}
