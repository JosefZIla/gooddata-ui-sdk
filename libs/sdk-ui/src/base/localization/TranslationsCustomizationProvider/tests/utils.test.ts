// (C) 2021 GoodData Corporation

import { pickCorrectInsightWording, removeAllInsightToReportTranslations } from "../utils";
import { IWorkspaceSettings } from "@gooddata/sdk-backend-spi";
import { ITranslations } from "../../../localization/IntlWrapper";

const mockTranslation: ITranslations = {
    "mock.translation|insight": "Insight",
    "mock.translation|report": "Report",
};

describe("pickCorrectInsightWording", () => {
    const settings: IWorkspaceSettings = {
        workspace: "workspace",
    };

    it.each([
        ["insight", false],
        ["report", true],
    ])(
        "should return translations with %s when enableInsightToReport is set to %s",
        (_value: string, enableInsightToReport: boolean) => {
            const result = pickCorrectInsightWording(mockTranslation, {
                ...settings,
                enableInsightToReport,
            });

            expect(result).toMatchSnapshot();
        },
    );
});

describe("removeAllInsightToReportTranslations", () => {
    const mockTranslationWithoutExtraWords: ITranslations = {
        "mock.translation": "Insight",
    };
    it("should remove all insight to report words", () => {
        const result = removeAllInsightToReportTranslations({
            ...mockTranslationWithoutExtraWords,
            ...mockTranslation,
        });
        expect(result).toMatchObject(mockTranslationWithoutExtraWords);
    });
});
