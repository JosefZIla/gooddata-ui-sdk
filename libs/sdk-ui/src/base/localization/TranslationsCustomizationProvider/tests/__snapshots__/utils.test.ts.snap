// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pickCorrectInsightWording should return translations with insight when enableInsightToReport is set to false 1`] = `
Object {
  "mock.translation": "Insight",
  "mock.translation|insight": "Insight",
  "mock.translation|report": "Report",
}
`;

exports[`pickCorrectInsightWording should return translations with report when enableInsightToReport is set to true 1`] = `
Object {
  "mock.translation": "Report",
  "mock.translation|insight": "Insight",
  "mock.translation|report": "Report",
}
`;
